enterprise/server/scheduling/priority_task_scheduler/priority_task_scheduler.go

```diff
@@ -557,6 +615,22 @@ func (q *PriorityTaskScheduler) Start() error {
 			q.handleTask()
 		}
 	}()
+
+	if *queueTrimInterval > 0 {
+		go func() {
+			timer := q.clock.NewTicker(*queueTrimInterval)
+			defer timer.Stop()
+
+			for {
+				select {
+				case <-q.rootContext.Done():
+					return
+				case <-timer.Chan():
+					q.trimQueue()
```

Comment:
if `TaskExists` takes a long time then `trimQueue` will keep spawning background goroutines that keep building up - I think we probably want the `TaskExists` calls to be done synchronously in this for-loop rather than in a bg goroutine

Reply:
Done